x/distribution/keeper/store.go

```diff
@@ -23,130 +20,25 @@ func (k Keeper) GetDelegatorWithdrawAddr(ctx context.Context, delAddr sdk.AccAdd
-	store := k.storeService.OpenKVStore(ctx)
-	bz, err := store.Get(types.ProposerKey)
-	if err != nil {
-		return nil, err
-	}
-
-	if bz == nil {
-		return nil, errors.New("previous proposer not set")
-	}
-
-	addrValue := gogotypes.BytesValue{}
-	err = k.cdc.Unmarshal(bz, &addrValue)
-	if err != nil {
-		return nil, err
-	}
-
-	return addrValue.GetValue(), nil
-}
-
-// set the proposer public key for this block
-func (k Keeper) SetPreviousProposerConsAddr(ctx context.Context, consAddr sdk.ConsAddress) error {
-	store := k.storeService.OpenKVStore(ctx)
-	bz := k.cdc.MustMarshal(&gogotypes.BytesValue{Value: consAddr})
-	return store.Set(types.ProposerKey, bz)
-}
-
-// historical reference count (used for testcases)
-func (k Keeper) GetValidatorHistoricalReferenceCount(ctx context.Context) (count uint64) {
-	store := runtime.KVStoreAdapter(k.storeService.OpenKVStore(ctx))
-	iter := storetypes.KVStorePrefixIterator(store, types.ValidatorHistoricalRewardsPrefix)
-	defer iter.Close()
-	for ; iter.Valid(); iter.Next() {
-		var rewards types.ValidatorHistoricalRewards
-		k.cdc.MustUnmarshal(iter.Value(), &rewards)
-		count += uint64(rewards.ReferenceCount)
-	}
-	return
-}
-
-// get slash event for height
-func (k Keeper) GetValidatorSlashEvent(ctx context.Context, val sdk.ValAddress, height, period uint64) (event types.ValidatorSlashEvent, found bool, err error) {
-	store := k.storeService.OpenKVStore(ctx)
-	b, err := store.Get(types.GetValidatorSlashEventKey(val, height, period))
-	if err != nil {
-		return types.ValidatorSlashEvent{}, false, err
-	}
-
-	if b == nil {
-		return types.ValidatorSlashEvent{}, false, nil
-	}
-
-	err = k.cdc.Unmarshal(b, &event)
-	if err != nil {
-		return types.ValidatorSlashEvent{}, false, err
-	}
-
-	return event, true, nil
-}
-
-// set slash event for height
-func (k Keeper) SetValidatorSlashEvent(ctx context.Context, val sdk.ValAddress, height, period uint64, event types.ValidatorSlashEvent) error {
-	store := k.storeService.OpenKVStore(ctx)
-	b, err := k.cdc.Marshal(&event)
-	if err != nil {
-		return err
-	}
-
-	return store.Set(types.GetValidatorSlashEventKey(val, height, period), b)
-}
-
 // iterate over slash events between heights, inclusive
 func (k Keeper) IterateValidatorSlashEventsBetween(ctx context.Context, val sdk.ValAddress, startingHeight, endingHeight uint64,
 	handler func(height uint64, event types.ValidatorSlashEvent) (stop bool),
-) {
-	store := runtime.KVStoreAdapter(k.storeService.OpenKVStore(ctx))
-	iter := store.Iterator(
-		types.GetValidatorSlashEventKeyPrefix(val, startingHeight),
-		types.GetValidatorSlashEventKeyPrefix(val, endingHeight+1),
-	)
-	defer iter.Close()
-	for ; iter.Valid(); iter.Next() {
-		var event types.ValidatorSlashEvent
-		k.cdc.MustUnmarshal(iter.Value(), &event)
-		_, height := types.GetValidatorSlashEventAddressHeight(iter.Key())
-		if handler(height, event) {
-			break
+) error {
+	rng := new(collections.Range[collections.Triple[sdk.ValAddress, uint64, uint64]]).
```

Comment:
Now that it is in collections we should probably just use the helper instead

Reply:
I think it's not the same, the other one is iterating all until a key. Here I'm iterating through values prefixed by an address, between 2 K2s